lib/Dialect/FIRRTL/FIRRTLOps.cpp

```diff
@@ -1374,6 +1381,28 @@ MemOp::PortKind MemOp::getPortKind(size_t resultNo) {
       getResult(resultNo).getType().cast<FIRRTLType>());
 }
 
+/// Return the number of bits in the mask for the memory.
+size_t MemOp::getMaskBits() {
+  assert(getNumResults() != 0 && "Mems with no read/write ports are illegal");
+
+  for (auto res : getResults()) {
+    auto firstPortType = res.getType().cast<FIRRTLType>();
+    if (getMemPortKindFromType(firstPortType) == PortKind::Read)
+      continue;
+
+    FIRRTLType mType;
+    for (auto t :
+         firstPortType.getPassiveType().cast<BundleType>().getElements()) {
+      if (t.name.getValue().contains("mask"))
+        mType = t.type;
+    }
+    if (mType.dyn_cast_or_null<UIntType>())
+      return mType.getBitWidthOrSentinel();
+  }
+  // Default is 1 bit.
+  return 1;
```

Comment:
should the default be 0?

Reply:
Right, if no write ports then mask defaults to 0. Fixed.